### ISSUE TITLE
Correct PDM Mic Routing for CLK and DATA

### ISSUE THREAD

**Issue by eyuan-creare** [Tympan/Tympan_Earpiece_Shield#4]
There are two issues related to the routing of the PDM CLK and DATA lines.

### 1) Earpiece Shield Header to Tympan Rev-E

Here is the header for the Tympan Rev-E (Rev-v2).  Note that CLK is on P3.1 and DATA is on P3.2
<img src="https://user-images.githubusercontent.com/42774675/135122242-97a04b76-7029-4ffe-b169-5ede9f21f076.png" height = 300/>

Here is the header for the Earpiece Shield (Rev-1).  Note that CLK and DAT are switched
<img src="https://user-images.githubusercontent.com/42774675/135122445-232a830d-0d9a-48c2-923f-ca40666574e0.png" height = 300/>

### 2) CLK and DATA signals on Second AIC

Here is the AIC on the Tympan Rev-E.  This is how the CLK and DATA signals should be connected.
<img src="https://user-images.githubusercontent.com/42774675/135123077-4a0bd4c8-1ecf-4bdc-a738-0205f907ef54.png" height = 100/>

Here is the AIC on the Earpiece Shield.  The CLK and DATA are reversed.
<img src="https://user-images.githubusercontent.com/42774675/135122956-77a5cfeb-f71e-4c6d-9e00-faa676343292.png" height = 100/>

I believe this all comes down to TI's poor choice of using a SCLK pin for PDM Data.

@biomurph Can you correct the labels on the earpiece shield and highlight these correction in a revision table?

------
In summary, the PDM signals from both AICs are swapped.  The only reason the existing earpieces work is because the signals were swapped on the rigid earpiece as well.  This shows the correct signal routing for the Rigid PCB.

<img src="https://user-images.githubusercontent.com/42774675/135123831-0bbf709a-d3c8-48c3-a74d-ff138609a237.png" height = 200/>

- I corrected the [Wiring Diagram_RIC Hearing Aid_Rev-D.pdf](https://github.com/Tympan/Earpiece_BTE-RIC_V1/tree/master/Electrical%20Design/Schematics_PDF) so that CLK and DATA are swapped.

- @biomurph Can you also change this on the [Rigid PCB](https://github.com/Tympan/Earpiece_BTE-RIC_V1/tree/master/Electrical%20Design/KICAD_BTE_RIGID)?

**Comment by eyuan-creare** [comment]
@biomurph Joel, remember this issue with swapped CLK and DAT lines?  Could you correct the schematic on the earpiece shield and leave a note that this was a bug, not a rerouting of lines?

Here is the current schematic.  The DAT and CLK lines are reversed.  Please check that.
<img width="1293" height="731" alt="Image" src="https://github.com/user-attachments/assets/caedd90f-e0b7-4b5b-950a-f484ef9e38b7" />

**Comment by biomurph** [comment]
@eyuan-creare I am switching the CLK and DAT lines on the following components

- P2 header
- U2 AIC
- J3 USB Mini
- J4 USB Mini

That is the extent of the changes, correct?

This will also require re-routing the PCB, which will happen in the future. I'll put a note in the Schematic and Board files.

**Comment by biomurph** [comment]
Actually, by swapping all occurrences of the DAT and CLK lines, I don't have to do any re-routing!
